bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/accessories/HomekitTemperatureSensorImpl.java

```diff
@@ -56,16 +58,23 @@ public void subscribeCurrentTemperature(HomekitCharacteristicChangeCallback call
 
     @Override
     public double getMinCurrentTemperature() {
-        return convertToCelsius(
+        // Apple defines default values in Celsius. We need to convert them to Fahrenheit if openHAB is using Fahrenheit
+        // convertToCelsius and convertFromCelsius are only converting if useFahrenheit is set to true, so no additional
+        // check here needed
+        return HomekitCharacteristicFactory.convertToCelsius(
                 getAccessoryConfiguration(HomekitCharacteristicType.CURRENT_TEMPERATURE, HomekitTaggedItem.MIN_VALUE,
-                        BigDecimal.valueOf(TargetTemperatureCharacteristic.DEFAULT_MIN_VALUE)).doubleValue());
+                        BigDecimal.valueOf(HomekitCharacteristicFactory
+                                .convertFromCelsius(CurrentTemperatureCharacteristic.DEFAULT_MIN_VALUE)))
```

Comment:
In several places, you replaced `TargetTemperatureCharacteristic.DEFAULT_MIN_VALUE` by `CurrentTemperatureCharacteristic.DEFAULT_MIN_VALUE`. Is it expected ?

Reply:
@lolodomo good catch.  but it was wrong before and CurrentTemperatureCharacteristic.DEFAULT_MIN_VALUE is correct one. 
there are 2 characteristics - current and target temperature and both of them have default min/max values defined by apple. 
in that method the current temperature is created so we need to use min/max values for current temperature. 
probably it has little or no impact at the end, but the CurrentTemperatureCharacteristic.DEFAULT_MIN_VALUE is correct one.

i should have mentioned this fix in the PR description.